### PR TITLE
Feature iOS 14 Widget

### PR DESCRIPTION
**Pull request by wesleybb**
## Description

This pull request contains changes to package management, updates to Swift 5 and iOS 14 widget support (small and medium). Bug fixing due to updating to iOS 14 and updating LocationTracker due to deprecated `authorizeStatus` function

I've tried to make a large widget that shows the sun location using SunView, however, I found out WidgetKit prevents too many updates. Converting UIVIew to View with UIViewRepresentable will not work in a Widget for that reason. Alternative methods I tried are screenshot the view after instantiating it or recreate it in View but with no success. I believe this is where my knowledge ends with Swift. In case anyone is interested to pick this up that would be amazing 🙏 

Now, the widget retrieves user location and time to show a message and location. Just like inside the app, and it changes the Theme color depending on the day.

## Type of Change
- [x] iOS 14 Widget (small, medium)
- [x] Fix authorizeStatus deprecation fix
- [x] Package manger from Carthage to Swift Package Manager
- [x] Fix share button
- [x] Convert to Swift 5
- [x] Fix presenting main modal to fullscreen 

## Proof that it works

image

P.S. I'm not a Swift developer, if you see any flaws or possible ways to improve the code, please feel free to make changes. The latest Testflight build that contains the latest update is 1.9.0 Build 58.

### CODE REVIEW

**Comment by 3lvis** [comment]
@wesleybb going crazy with iOS?! I digg this. 👏

**Comment by designatednerd** [comment]
one of us
one of us
one of us

(also in actual advice you should be able to git ignore the `.build` folder and save yourself a whole mess of LOC)

File: DaylightWidget/WidgetTextView.swift [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+// View for showing weather message and country
+struct WidgetTextView: View {
+    var entry: Provider.Entry
+    
+    var body: some View {
+        ZStack {
+            Color(entry.daylightController.primaryColor).ignoresSafeArea()
+            
+            VStack(alignment: .center, spacing: 12, content: {
+                Text(entry.daylightController.stringMessage!).multilineTextAlignment(.center).font(Font(Theme.light(size: 15)))
```
Comment (designatednerd) [inline DaylightWidget/WidgetTextView.swift:19]: a) Generally modifiers are done on separate lines such as:

```swift
Text(entry.daylightController.stringMessage!)
  .multilineTextAlignment(.center)
  .font(Font(Theme.light(size: 15)))
```

b) I would SUPER not recommend force unwrapping `stringMessage` here. Remember that `!` basically means crash if this isn't here. One thing you can do is use `entry.daylightController.stringMessage ?? " placeholder"` to give it a default value if the message isn't there.

File: Sources/daylight-ios/daylight_ios.swift [hunk 2]
```diff
@@ -0,0 +1,3 @@
+struct daylight_ios {
+    var text = "Hello, World!"
+}
```
Comment (designatednerd) [inline Sources/daylight-ios/daylight_ios.swift:3]: What's going on in this file?

File: iOS/Source/MainViewController.swift [hunk 3]
```diff
@@ -1,10 +1,12 @@
-import SweetUIKit
+//import SweetUIKit
 import UIKit
 import TinyConstraints
 
 class MainViewController: UIViewController {
     let insets = UIEdgeInsets(top: 40, left: 40, bottom: 40, right: 40)
+    let screenSize = UIScreen.main.bounds.size
```
Comment (designatednerd) [inline iOS/Source/MainViewController.swift:7]: unless i'm mistaken, I don't think this is actually used

File: DaylightWidget/DaylightWidget.swift [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+    
+    func placeholder(in context: Context) -> SimpleEntry {
+        let location = Location(coordinate: CLLocationCoordinate2D(latitude: 59.8937806, longitude: 10.6450379), city: "Oslo", country: "Norway")
+        let controller = DaylightModelController(location: location)
+        
+        return SimpleEntry(date: Date(), daylightController: controller)
+    }
+
+    func getSnapshot(in context: Context, completion: @escaping (SimpleEntry) -> ()) {
+        let location = Location(coordinate: CLLocationCoordinate2D(latitude: 59.8937806, longitude: 10.6450379), city: "Oslo", country: "Norway")
```
[the remark this answers is not in the document]
Comment (wmcheung) [reply DaylightWidget/DaylightWidget.swift:24]: I don't think it is possible to get the user location here. getSnapshot shows you a preview of how the widget could look like with dummy data. Before adding it to the home screen.

[the remark this answers is not in the document]
Comment (designatednerd) [reply DaylightWidget/DaylightWidget.swift:24]: got it! Yeah then I would say maybe just have that be more centralized

File: Sources/daylight-ios/daylight_ios.swift (hunk 2 above)
Comment (wesleybb) [reply Sources/daylight-ios/daylight_ios.swift:3]: No idea how this got here, will remove it

**Pull request merged**